Cmfcmf/OpenWeatherMap.php

```diff
@@ -668,6 +678,67 @@ private function buildUVIndexUrl($mode, $lat, $lon, $cnt = null, \DateTime $star
         return sprintf($this->uvIndexUrl . '%s?%s', $requestMode, http_build_query($params));
     }
 
+    public function getCoordinatesByLocationName(
+        $city,
+        $stateCode = null,
+        $countryCode = null,
+        $limit = null
+    ) {
+        $url = $this->buildCoordinatesByLocationNameUrl($city, $stateCode, $countryCode, $limit);
+        $response = $this->cacheOrFetchResult($url);
+        $data = $this->parseJson($response);
+        $cities = [];
+        foreach ($data as $datum) {
+            $cities[] = new City(
+                -1,
+                $datum->name,
+                $datum->lat,
+                $datum->lon,
+                $datum->country,
+                null,
+                null,
+                $datum->state,
+                (array)$datum->local_names
+            );
```

Comment:
Could you create a new `City` class specifically for this API call? I don't like how the newly added fields are only used by this API, and how this API never sets some of the fields.  Maybe call it `Location`?